### PR TITLE
Node package and more!

### PR DESCRIPTION
**Pull request by VikingTristan**
- Rework and update webpack and pave way for publishing a node package
- Update Design Guide script itself to be easier to import from others.
- Make our hosted Design Guide slightly different ( the "cdn" one )
  - The cdn version will autoload itself, the imported one from node modules will not.
- Update jest with new major and all tests accordingly
- Upgrade all node packages except chart.js ( it has a new major and requires a PR by itself )
- Add a workflow for publishing a node package whenever there is a new tag and version in the package.json.
- Add new documentation for installation and usage of new node package.
- Add a little message when users install package

image

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have updated the **CHANGELOG** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Review by asbjornu — CHANGES_REQUESTED** [review]
Wonderful!

File: .github/workflows/node-package.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        with:
+          node-version: 16
+
+      - name: Cache dependencies
+        uses: actions/cache@v2.1.6
+        with:
+          path: ~/.npm
+          key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-node-
```
[the remark this answers is not in the document]
Comment (VikingTristan) [reply .github/workflows/node-package.yml:23]: ooooooo, very nice.

File: README.md [hunk 2]
```diff
@@ -11,6 +11,38 @@
 
 ## Installing / Getting started
 
+```sh
+npm i @swedbankpay/design.swedbankpay.com
```
[the remark this answers is not in the document]
Comment (VikingTristan) [reply README.md:15]: agree

File: README.md [hunk 3]
```diff
@@ -11,6 +11,38 @@
...
+sheet.init();
+```
+
+#### CSS
+```css
+@import url("@swedbankpay/design.swedbankpay.com/dist/designguide/styles/swedbankpay.css");
+```
+
+#### Assets
+When using the node package you also need to host assets such as fonts and icons. Copy the files at `@swedbankpay/design.swedbankpay.com/dist/designguide` to your static files.
```
[the remark this answers is not in the document]
Comment (VikingTristan) [reply README.md:39]: Hmm, interesting...

File: package.json [hunk 4]
```diff
@@ -1,103 +1,108 @@
...
-        "stylelint-declaration-block-no-ignored-properties": "^2.3.0",
-        "terser-webpack-plugin": "^5.1.4",
-        "testcafe": "^1.15.0",
-        "webpack": "^5.44.0",
-        "webpack-bundle-analyzer": "^4.4.2",
-        "webpack-cli": "^4.7.2",
-        "webpack-dev-server": "^3.11.2"
-    }
+  "name": "@swedbankpay/design.swedbankpay.com",
+  "version": "6.1.0",
```
[the remark this answers is not in the document]
Comment (VikingTristan) [reply package.json:3]: that's an interesting idea

File: src/App/ComponentsDocumentation/components/Charts/Introduction/index.js [hunk 5]
```diff
@@ -6,8 +6,6 @@ import CodeTags from "@components/CodeTags";
 const basename = process.env.basename;
 const scriptUrl = `${window.location.origin + basename}scripts/dg.js`;
 const dashboardScriptUrl = `${window.location.origin + basename}scripts/dg-dashboard.js`;
-const styleUrl = `${window.location.origin + basename}styles/dg-style.css`;
```
[the remark this answers is not in the document]
Comment (VikingTristan) [reply src/App/ComponentsDocumentation/components/Charts/Introduction/index.js:9]: As I was updating the url to point to new css files, I noticed that the CSS isn't being used at all in charts. So I removed it entirely.

File: README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (VikingTristan) [reply README.md]: I wrote your suggestion as a suggestion in the doc :)

File: package.json (hunk 4 above)
[the remark this answers is not in the document]
Comment (VikingTristan) [reply package.json:3]: very interesting, but to be honest I'm not entirely sure how to proceed with this and I would like to perhaps do it in its own PR in the future

[the remark this answers is not in the document]
Comment (asbjornu) [reply package.json:3]: Sure, good idea.

**Review by asbjornu — APPROVED** [review]
👏🏼 👏🏼 👏🏼

**Comment by VikingTristan** [comment]
@asbjornu gah, noticed some minor rendering bugs with new imports on a few pages.
incoming fixes throughout the day

**Pull request merged**